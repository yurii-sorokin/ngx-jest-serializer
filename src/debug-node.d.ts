export declare class DebugNodeEventListener {
  name: string
  callback: Function
}

export interface DebugNodeProperties {
  [key: string]: any
}

export interface DebugNodeAttributes {
  [key: string]: string | null
}

export interface DebugNodeClasses {
  [key: string]: boolean
}

export interface DebugNodeStyles {
  [key: string]: string | null
}

export interface DebugNodeReferences {
  [key: string]: any
}

export interface DebugNode {
  _debugContext: any
  name: string
  nativeNode: Node
  childNodes: DebugNode[]
  listeners: DebugNodeEventListener[]
  properties: DebugNodeProperties
  attributes: DebugNodeAttributes
  classes: DebugNodeClasses
  styles: DebugNodeStyles
  parent: any
  readonly injector: any
  readonly componentInstance: any
  readonly context: any
  readonly references: DebugNodeReferences
  readonly providerTokens: any[]
}
