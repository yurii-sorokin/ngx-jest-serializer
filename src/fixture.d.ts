import { DebugNode } from './debug-node'

export declare class ComponentFixture<T = any> {
  componentRef: any
  ngZone: any | null
  debugElement: DebugNode
  componentInstance: T
  nativeElement: any
  elementRef: any
  changeDetectorRef: any
}
