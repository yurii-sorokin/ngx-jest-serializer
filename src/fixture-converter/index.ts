export * from './fixture-to-json'
export * from './debug-node-to-json'
