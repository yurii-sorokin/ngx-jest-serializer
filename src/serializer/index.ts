export * from './default-serializer'
export * from './fixture-serializer-factory'
