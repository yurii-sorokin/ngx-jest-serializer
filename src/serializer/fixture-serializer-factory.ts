import { Plugin } from 'pretty-format'
import { fixtureToJSON, DebugNodeConverterOptions } from '../fixture-converter'
import { ComponentFixture } from '../fixture'

const defaultOptions: Partial<DebugNodeConverterOptions> = {
  printComments: false,
  ignoreProps: ['ng-version']
}

function isAngularFixture(fixture: ComponentFixture): boolean {
  return Boolean(fixture && fixture.componentRef)
}

function createFixtureSerializer(options: DebugNodeConverterOptions = {}): Plugin {
  return {
    print(value: any, serialize: (data: any) => any) {
      return serialize(fixtureToJSON(value, { ...defaultOptions, ...options }))
    },
    test(value: any) {
      return isAngularFixture(value)
    }
  }
}

export {
  createFixtureSerializer
}
