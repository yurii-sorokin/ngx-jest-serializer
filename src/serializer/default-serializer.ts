import { Plugin } from 'pretty-format'
import { createFixtureSerializer } from './fixture-serializer-factory'

export const defaultFixtureSerializer: Plugin = createFixtureSerializer()
