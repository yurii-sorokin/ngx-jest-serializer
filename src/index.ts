import { defaultFixtureSerializer } from './serializer'

export = defaultFixtureSerializer
