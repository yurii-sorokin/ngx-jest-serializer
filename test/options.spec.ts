import { ComponentFixture } from '@angular/core/testing'
import { createFixtureSerializer } from '../src/serializer'
import { initHostFixture } from './framework/host'

describe(`Options`, () => {
  let fixture: ComponentFixture<any>

  beforeEach(() => {
    fixture = initHostFixture({
      template:
        `<div
          [someInput]="'some input static value'"
          [anotherInput]="'another input static value'"
          [ngClass]="{ 'directiveClassOn': true, 'directiveClassOff': false }"
          [ngStyle]="{ 'directiveStyleRule': 'directiveStyleRuleValue' }"
          (someListener)="someAction()"
          (anotherListener)="anotherAction()"
          class="staticClass"
          style="staticStyleRule: staticStyleRuleValue;"
          someAttr="some attr value"
          anotherAttr="another attr value"
          data-some-attr="some data attr value"
          data-another-attr="another data attr value"
        >
          <!--static comment-->
          <ng-container *ngIf="true"></ng-container>
          <ng-container *ngIf="false"></ng-container>
          <div *ngIf="true">conditional</div>
          <div *ngIf="false"></div>
        </div>`,
      props: {
        someAction: () => 'do something',
        anotherAction: () => 'do something else'
      },
      host: {
        '[boundAttr]': `'bound attr value'`,
        '[class.boundClass]': `true`,
        '[style.display]': `'block'`,
        '(boundListener)': `boundAction`,
      }
    })
  })

  it(`should ignore "ng-version" property by default`, () => {
    expect.addSnapshotSerializer(createFixtureSerializer({}))

    expect(fixture).toMatchSnapshot()
  })

  it(`should ignore specified properties`, () => {
    expect.addSnapshotSerializer(createFixtureSerializer({
      ignoreProps: [
        'boundAttr',
        'boundListener',
        'class',
        'style',
        'someInput',
        'someAttr',
        'data-some-attr',
        'someListener',
      ]
    }))

    expect(fixture).toMatchSnapshot()
  })

  it(`should ignore properties by regular expression`, () => {
    expect.addSnapshotSerializer(createFixtureSerializer({
      ignoreProps: [
        '/.*some/',
        '/^ng-reflect/',
      ]
    }))

    expect(fixture).toMatchSnapshot()
  })

  it(`should not preserve binding comments by default`, () => {
    expect.addSnapshotSerializer(createFixtureSerializer())

    expect(fixture).toMatchSnapshot()
  })

  it(`should not preserve binding comments if option is disabled`, () => {
    expect.addSnapshotSerializer(createFixtureSerializer({
      printComments: false
    }))

    expect(fixture).toMatchSnapshot()
  })

  it(`should preserve binding comments if option is enabled`, () => {
    expect.addSnapshotSerializer(createFixtureSerializer({
      printComments: true
    }))

    expect(fixture).toMatchSnapshot()
  })
})
