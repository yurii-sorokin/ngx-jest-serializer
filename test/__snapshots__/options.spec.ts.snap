// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options should ignore "ng-version" property by default 1`] = `
<tst-host
  boundAttr="bound attr value"
  boundListener={[Function]}
  class="boundClass"
  style="display: block"
>
  <div
    anotherAttr="another attr value"
    anotherInput="another input static value"
    anotherListener={[Function]}
    class="staticClass staticClass directiveClassOn"
    data-another-attr="another data attr value"
    data-some-attr="some data attr value"
    ng-reflect-klass="staticClass"
    ng-reflect-ng-class="[object Object]"
    ng-reflect-ng-style="[object Object]"
    someAttr="some attr value"
    someInput="some input static value"
    someListener={[Function]}
    style="directiveStyleRule: directiveStyleRuleValue"
  >
    
          
    
          
    
          
    
          
    <div>
      conditional
    </div>
    
          
    
        
  </div>
</tst-host>
`;

exports[`Options should ignore properties by regular expression 1`] = `
<tst-host
  boundAttr="bound attr value"
  boundListener={[Function]}
  class="boundClass"
  ng-version="4.4.6"
  style="display: block"
>
  <div
    anotherAttr="another attr value"
    anotherInput="another input static value"
    anotherListener={[Function]}
    class="staticClass staticClass directiveClassOn"
    data-another-attr="another data attr value"
    style="directiveStyleRule: directiveStyleRuleValue"
  >
    
          
    
          
    
          
    
          
    <div>
      conditional
    </div>
    
          
    
        
  </div>
</tst-host>
`;

exports[`Options should ignore specified properties 1`] = `
<tst-host
  ng-version="4.4.6"
>
  <div
    anotherAttr="another attr value"
    anotherInput="another input static value"
    anotherListener={[Function]}
    data-another-attr="another data attr value"
    ng-reflect-klass="staticClass"
    ng-reflect-ng-class="[object Object]"
    ng-reflect-ng-style="[object Object]"
  >
    
          
    
          
    
          
    
          
    <div>
      conditional
    </div>
    
          
    
        
  </div>
</tst-host>
`;

exports[`Options should not preserve binding comments by default 1`] = `
<tst-host
  boundAttr="bound attr value"
  boundListener={[Function]}
  class="boundClass"
  style="display: block"
>
  <div
    anotherAttr="another attr value"
    anotherInput="another input static value"
    anotherListener={[Function]}
    class="staticClass staticClass directiveClassOn"
    data-another-attr="another data attr value"
    data-some-attr="some data attr value"
    ng-reflect-klass="staticClass"
    ng-reflect-ng-class="[object Object]"
    ng-reflect-ng-style="[object Object]"
    someAttr="some attr value"
    someInput="some input static value"
    someListener={[Function]}
    style="directiveStyleRule: directiveStyleRuleValue"
  >
    
          
    
          
    
          
    
          
    <div>
      conditional
    </div>
    
          
    
        
  </div>
</tst-host>
`;

exports[`Options should not preserve binding comments if option is disabled 1`] = `
<tst-host
  boundAttr="bound attr value"
  boundListener={[Function]}
  class="boundClass"
  style="display: block"
>
  <div
    anotherAttr="another attr value"
    anotherInput="another input static value"
    anotherListener={[Function]}
    class="staticClass staticClass directiveClassOn"
    data-another-attr="another data attr value"
    data-some-attr="some data attr value"
    ng-reflect-klass="staticClass"
    ng-reflect-ng-class="[object Object]"
    ng-reflect-ng-style="[object Object]"
    someAttr="some attr value"
    someInput="some input static value"
    someListener={[Function]}
    style="directiveStyleRule: directiveStyleRuleValue"
  >
    
          
    
          
    
          
    
          
    <div>
      conditional
    </div>
    
          
    
        
  </div>
</tst-host>
`;

exports[`Options should preserve binding comments if option is enabled 1`] = `
<tst-host
  boundAttr="bound attr value"
  boundListener={[Function]}
  class="boundClass"
  style="display: block"
>
  <div
    anotherAttr="another attr value"
    anotherInput="another input static value"
    anotherListener={[Function]}
    class="staticClass staticClass directiveClassOn"
    data-another-attr="another data attr value"
    data-some-attr="some data attr value"
    ng-reflect-klass="staticClass"
    ng-reflect-ng-class="[object Object]"
    ng-reflect-ng-style="[object Object]"
    someAttr="some attr value"
    someInput="some input static value"
    someListener={[Function]}
    style="directiveStyleRule: directiveStyleRuleValue"
  >
    
          
    
          
    <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
    <!---->
    
          
    <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
    
          
    <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
    <div>
      conditional
    </div>
    
          
    <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
    
        
  </div>
</tst-host>
`;
